horace_core/sqw/@sqw/sqw.m

```diff
@@ -164,17 +140,84 @@
         varargout = resolution_plot (w, varargin);
         wout = noisify(w,varargin);
         %----------------------------------
+    end
+    %======================================================================
+    % METHODS, Available on SQW but redirectiong actions to DnD and requesting
+    % only DND object for implementation.
+    methods
+        function [nd,sz] = dimensions(win)
+            % Return size and shape of the image
+            % arrays in sqw or dnd object
+            [nd,sz] = win(1).data.dimensions();
+        end
+        %
+        function [val, n] = data_bin_limits (obj)
+            % Get limits of the data in an n-dimensional dataset, that is,
+            % find the coordinates along each of the axes of the smallest
+            % cuboid that contains bins with non-zero values of
+            % contributing pixels.
+            %
+            % Syntax:
+            %   >> [val, n] = data_bin_limits (din)
+            %
+            [val,n] = obj.data.data_bin_limits();
+        end
+        %
+        wout = smooth(win, varargin)  % smooth sqw object or array of sqw
```

Comment:
presuming this is a declaration of a method contained in a separate file. It needs blank lines round it and moved to the top of the methods section